### DISCUSSION TITLE
Wix 5 - Always install prerequisites (AlwaysInstallPrereqs)

### DISCUSSION THREAD

**Discussion by jirifa** [wixtoolset/issues#8609] [Questions]
How to force Wix 5 to always invoke Prereq BA to install all Prerequisites?

In Wix 4, there is an attribute `bal:AlwaysInstallPrereqs="yes"` on the `WixDotNetCoreBootstrapperApplicationHost` element.
In Wix 5, there is only `WixPrerequisiteBootstrapperApplication` element with no such attribute.

The thing is, that we need to install .NET Framework 4.8, .NET 8 Hosting Bundle and .NET 8 SDK .
Our BA is capable to run only with .NET 8 Desktop Runtime.

It seems, that the Prereq BA is invoked only when .NET 8 Desktop Runtime is missing on the machine.
When the Desktop Runtime is installed, Prereq BA won't start. But it doesn't mean, that Framework 4.8, Hosting bundle and SDK are installed!

Does anyone know how to solve it?

**Comment by michaelgburton** [comment]
There are a few different things that you might need to do here.

First, make sure your custom BA is marked secondary:

`<BootstrapperApplication Name="MyInstallerBA" Secondary="true" SourceFile="[PATH_TO_BA] />`

		
Then, make sure the executable package(s) that you need to install should be marked as PrereqPackage="true", for example:

```
<ExePackage
	bal:PrereqPackage="yes" 
	Id="NET8_X86_DESKTOP"
	DisplayName="Microsoft Windows Desktop Runtime - 8.0.8 (x86)"
	Description="Microsoft Windows Desktop Runtime - 8.0.8 (x86)"
	InstallCondition="VersionNT"
	DetectCondition="(NET8_INSTALLED_VER &gt;= NET8_VER) AND VersionNT"
	Permanent ="yes"
	Protocol="burn"
	PerMachine="yes" 
	Cache="remove" 
	SourceFile="[PATH_TO_SOURCE_FILE]"/>
```

Finally, make sure your `InstallCondition` and `DetectCondition` on the package are evaluating to `true` and `false`, respectively.

There is a bunch of stuff in the development backlog that talks about only supporting a single prereq package, but I get the impression that WiX 4 may have relaxed this requirement. I'm not 100% certain whether this is the case, as our BA only requires the one prerequisite package, but I think you should try the above, with all of your prereq packages set up as described above.
